arch/Makefile.darwin_arm64_gfortran

```diff
@@ -44,8 +44,8 @@ DEBUG =
 OPTIM = -O3
 
 DEFINES += -DFORTRAN_UNDERSCORE  -DGETARG_F2003 -DGETENV_F2003 -DGFORTRAN -DDARWIN -fPIC
-F95FLAGS += -x f95-cpp-input -ffree-line-length-none -ffree-form  -fno-second-underscore  -fPIC
-F77FLAGS += -x f77-cpp-input -fno-second-underscore -fPIC
+F95FLAGS += -x f95-cpp-input -ffree-line-length-none -ffree-form  -fno-second-underscore  -fPIC -fno-realloc-lhs
```

Comment:
Do we need to make an analogous change for intel compilers?

Reply:
Would be good for consistency I think but should work either way.

Reply:
The ifort equivalent seems to be `-nostandard-realloc-lhs`.

Reply:
OK let's add that. Intel CI is WIP: https://github.com/libAtoms/QUIP/pull/329